### PR TITLE
Upgrade OpenROAD

### PR DESCRIPTION
**Pull request by mithro**
This pull request includes;
 * #197 
 * #186 
 * #201

Adds `ortools`.

### CODE REVIEW

File: dependency_support/org_theopenroadproject/org_theopenroadproject.bzl [hunk 1]
```diff
@@ -21,15 +21,9 @@ def org_theopenroadproject():
     maybe(
         new_git_repository,
         name = "org_theopenroadproject",
-        commit = "21b84678f4eb0eae2e38b897e4e0d36e145f99cd",
+        commit = "65053db0233ffb6e3235b3f28774afbcd063d569",
```
Comment (hzeller) [inline dependency_support/org_theopenroadproject/org_theopenroadproject.bzl:24]: Can we merge past commit https://github.com/The-OpenROAD-Project/OpenROAD/commit/6eab677861f92ec4699ddc2233c7b57ef701778e ?

**Comment by mithro** [comment]
I'm unsure if there is a better way to do the `eigen` change and the `ortools` include patch.

**Comment by mithro** [comment]
Seems to be currently failing due to;
```
Step #1: Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
Step #1: In file included from external/org_theopenroadproject/src/odb/src/db/dbTechMinCutOrAreaRule.cpp:41:
Step #1: external/org_theopenroadproject/src/odb/include/odb/lefout.h:35:10: fatal error: 'boost/polygon/polygon.hpp' file not found
Step #1: #include <boost/polygon/polygon.hpp>
Step #1:          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
Step #1: 1 error generated.
Step #1: INFO: Elapsed time: 1504.421s, Critical Path: 123.36s
Step #1: INFO: 6047 processes: 3718 remote cache hit, 1755 internal, 572 linux-sandbox, 2 local.
Step #1: FAILED: Build did NOT complete successfully
Step #1: INFO: Streaming build results to: https://app.buildbuddy.io/invocation/cb596038-6263-11ee-bd15-0242c0a80a02
Step #1: FAILED: Build did NOT complete successfully
```

**Comment by QuantamHD** [comment]
The dependency support test is weird it's also failing for Eric. I don't see it defined anywhere though

**Comment by QuantamHD** [comment]
LGTM

File: dependency_support/com_google_ortools/com_google_ortools.bzl [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        remote = "https://gitlab.com/libeigen/eigen.git",
+        build_file_content = """
+cc_library(
+    name = 'eigen3',
+    srcs = [],
+    includes = ['.'],
+    hdrs = glob(['Eigen/**']),
+    visibility = ['//visibility:public'],
+)
+""",
```
Comment (QuantamHD) [inline dependency_support/com_google_ortools/com_google_ortools.bzl:60]: We do have an eigen already is that one not compatible with ortools?

Comment (mithro) [reply dependency_support/com_google_ortools/com_google_ortools.bzl:60]: I wasn't able to figure out how to map the ortools dependency to the one we already had.

**Pull request merged**